kite-data/kite-data-core/src/main/java/org/kitesdk/data/DatasetException.java

```diff
@@ -46,4 +46,11 @@ public DatasetException(Throwable t) {
     super(t);
   }
 
+  protected static String format(String message, Object... args) {
+    String[] argStrings = new String[args.length];
+    for (int i = 0; i < args.length; i += 1) {
+      argStrings[i] = String.valueOf(args[i]);
+    }
+    return String.format(String.valueOf(message), (Object[]) argStrings);
```

Comment:
I think this can all be achieved by making the contents of this method `String.format(message, args)` since that will take care of formatting null (e.g. when you use `%s`).

Not a big deal either way though - this change shouldn't block the overall merge.